src/DataDecoder.php

```diff
@@ -33,13 +42,25 @@ class DataDecoder
     private $dataDecodingStrategy;
 
     /**
-     * Set data decoding strategy.
+     * Set format of decoded data.
+     *
+     * @param string $dataFormat
+     */
+    public function setDataFormat($dataFormat)
```

Comment:
This function doesn't set data format only. It also sets up deciding strategy, so it does more than promises.
Maybe name setUpWithDataFormat would be better.